lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm

```diff
@@ -42,19 +45,84 @@ sub doInventory {
     # Get agent capabilities
     my @agentCapabilities = _listAgentCapabilities();
 
+    my $customProperties = _getCustomProperties(logger => $logger);
+
     my $rudder = {
         HOSTNAME => $hostname,
         UUID => $Uuid,
         AGENT => \@agents,
         SERVER_ROLES => { SERVER_ROLE => \@serverRoles },
         AGENT_CAPABILITIES => { AGENT_CAPABILITY => \@agentCapabilities },
+        CUSTOM_PROPERTIES => $customProperties,
     };
 
     $inventory->addEntry(
         section => 'RUDDER', entry => $rudder
     );
 }
 
+sub _getCustomProperties {
+    my (%params) = @_;
+    my $logger   = $params{logger};
+
+    my $custom_properties_dir = ($OSNAME eq 'MSWin32') ? 'C:\Program Files\Rudder\hooks.d' : '/var/rudder/hooks.d';
+    my $custom_properties;
+    if (-d "$custom_properties_dir") {
+        my @custom_properties_list = ();
+        my @ordered_script_list = ();
+        opendir(DIR, $custom_properties_dir);
+        # List each file in the custom_properties directory, each files being a script
+        @ordered_script_list = sort readdir(DIR);
+        closedir(DIR);
+        while (my $file = shift @ordered_script_list) {
+            my $script_file = $custom_properties_dir . "/" . $file;
+            if (-f $script_file) {
+                next if ($file =~ m/^\./);
+                # Ignore non-executable file, or folders
+                next unless -x $script_file;
+
+                # Check that the file is not world writable
```

Comment:
here, I failed to check if file was owned by root (or its Windows equivalent) - as there can be several admin
how would you do that ? @peckpeck @g-bougard

Reply:
I used 
my $stats = stat($script_file);
my $owner = $stats->uid;
my $currentUser = $<;

 file must be owned by root or current user
if (($owner != 0) && ($owner != $currentUser)) {
                    $logger->error("Skipping script $script_file as it is not owned by root nor current user (owner is $owner)") if $logger;
                   next;
}
but this fails miserably on Windows - it always believe the owner of the file is 0